tests/test_priority_samples_process.py

```diff
@@ -0,0 +1,379 @@
+
+        self.mock_add_dart_plate.side_effect = find_plate_status
+
+        def extract_mongo_record(info: RecordScenarioList) -> SampleDocValue:
+            return info[0]
+
+        # Set expected mlwh samples
+        self.expected_mlwh_samples = list(
+            map(
+                extract_mongo_record,
+                filter(lambda info: info[1], zip(testing_priority_samples, request.param["expected_mlwh_samples"])),
+            )
+        )
+
+        # Set expected dart samples
+        self.expected_dart_samples = list(
+            map(
+                extract_mongo_record,
+                filter(lambda info: info[1], zip(testing_samples, request.param["expected_dart_samples"])),
+            )
+        )
+
+        # Set expected dart plates
+        self.expected_dart_plates = request.param["expected_dart_plates"]
+
+    # Currently not being ran
+    def validate_expected_data(self, with_different_scenarios, testing_priority_samples, testing_samples):
+        # Check list of expected mlwh samples
+        # Get unprocessed samples from list
+        def extract_priority_sample(info: RecordScenarioList) -> SampleDocValue:
+            return info[0]
+
+        expected_mlwh_samples = list(
+            map(
+                extract_priority_sample,
+                filter(lambda info: not (info[1]), zip(testing_priority_samples, self.processed_status)),
+            )
+        )
+        assert self.expected_mlwh_samples == expected_mlwh_samples
+        # Check list of expected dart samples and plates
+        result = list(
+            filter(
+                lambda info: ((self.plate_status[info[1]] == DART_STATE_PENDING) and not (info[2])),
+                zip(testing_samples, self.plate_barcodes, self.processed_status),
+            )
+        )
+        if len(result) == 0:
+            assert self.expected_dart_samples == []
+            assert self.expected_dart_plates == set(self.plate_barcodes)
+        else:
+            expected_dart_samples, expected_dart_plates, _ = zip(*result)
+            expected_dart_plates = set(expected_dart_plates)
+            assert self.expected_dart_samples == list(expected_dart_samples)
+            assert self.expected_dart_plates == list(expected_dart_plates)
+
+    def test_when_for_one_priority_sample_doesnt_exist_the_related_sample(
+        self, mongo_database, config, mlwh_connection, with_different_scenarios
+    ):
+        # Creates one error sample priority
+        _, mongo_database = mongo_database
+        collection = get_mongo_collection(mongo_database, COLLECTION_PRIORITY_SAMPLES)
+        _id = collection.find({})[0]["_id"]
+        collection.find_one_and_update({"_id": _id}, {"$set": {"sample_id": "aaaaaaaxxxaaaaaaaaaaaaa1"}})
+
+        try:
+            update_priority_samples(mongo_database, config, True)
+        except Exception:
+            pytest.fail("Unexpected error ..")
+
+    def test_mlwh_was_correctly_updated_in_update_priority_samples(
```

Comment:
This test feels complex, is it possible to refactor into two or reduce complexity somehow?